tutorials/transparent-proxy-and-filtering-on-k8s/index.md

```diff
@@ -0,0 +1,248 @@
+---
+title: Transparent Proxy and Filtering on Kubernetes
+description: Learn how to run a transparent proxy on Kubernetes to filter and intercept traffic out of your deployments.
+author: danisla
+tags: Kubernetes mitmproxy proxy
+date_published: 2017-09-23
+---
+
+Dan Isla | Google Cloud Solution Architect | Google
+
+There are many application environments and scenarios where you may want to filter and intercept all HTTP/S traffic out of a pod. Example use cases include isolating external access to specific HTTP or HTTPS paths, methods and in-flight altering of requests.
+
+An example of HTTPS interception would be filtering access to Google Cloud Storage buckets. Most proxy solutions do not support HTTPS inspection so the filtering that can be done is limited at the IP/DNS level. For the case of a Cloud Storage bucket, the bucket name is embedded under the https://storage.googleapis.com/BUCKET_NAME/object URL so filtering is limited to all buckets under the storage.googleapis.com domain, not any specific bucket.
+
+Another example would be the in-flight modification of requests. If you wanted to transparently add request/response headers to requests between services for tracing purposes, this can be done without the service having to explicitly set them when the request is made.
+
+[Mitmproxy](https://mitmproxy.org/) is an Open Source tool that can be used to intercept and modify HTTP and HTTPS requests transparently using the Python scripting language.
```

Comment:
Suggestion:
If you want to transparently add request/response headers to requests between services for tracing purposes, you can do that without the services having to explicitly set them.